src/util/IpcQueue.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <memory>
+#include <optional>
+
+class QByteArray;
+class QString;
+
+namespace chatterino::ipc {
+
+void sendMessage(const char *name, const QByteArray &data);
+
+class IpcQueuePrivate;
+class IpcQueue
```

Comment:
It would be better if IpcQueue could only have a successful state.
We can achieve this by not exposing the constructor, but only a "creator" like the `tryReplaceOrCreate` function, that returns a unique_ptr to an IpcQueue. It will just need to return a nullptr if it failed.

As we talked about in chat, maybe

```cpp
std::pair<std::unique_ptr<IpcQueue>, QString>
```

so it can be used like this:

```c++
auto [messageQueue, messageQueueError] = IpcQueue::create(...);
if (!messageQueue) {
  // handle error and print the error or something
}
```